datafusion/expr/src/window_frame.rs

```diff
@@ -479,8 +498,123 @@ mod tests {
+                ast::WindowFrameBound::Preceding($value),
+                &ast::WindowFrameUnits::$unit,
+            )?;
+            assert_eq!(preceding, WindowFrameBound::Preceding($expected));
+            let following = WindowFrameBound::try_parse(
+                ast::WindowFrameBound::Following($value),
+                &ast::WindowFrameUnits::$unit,
+            )?;
+            assert_eq!(following, WindowFrameBound::Following($expected));
+        };
+    }
+
+    macro_rules! test_bound_err {
+        ($unit:ident, $value:expr, $expected:expr) => {
+            let err = WindowFrameBound::try_parse(
+                ast::WindowFrameBound::Preceding($value),
+                &ast::WindowFrameUnits::$unit,
+            )
+            .unwrap_err();
+            assert_eq!(err.strip_backtrace(), $expected);
+            let err = WindowFrameBound::try_parse(
+                ast::WindowFrameBound::Following($value),
+                &ast::WindowFrameUnits::$unit,
+            )
+            .unwrap_err();
+            assert_eq!(err.strip_backtrace(), $expected);
+        };
+    }
+
+    #[test]
+    fn test_window_frame_bound_creation() -> Result<()> {
+        //  Unbounded
+        test_bound!(Rows, None, ScalarValue::Null);
+        test_bound!(Groups, None, ScalarValue::Null);
+        test_bound!(Range, None, ScalarValue::Null);
+
+        // Numeric (Number)
+        let number = Some(Box::new(ast::Expr::Value(ast::Value::Number(
+            "42".to_string(),
+            false,
+        ))));
+        test_bound!(Rows, number.clone(), ScalarValue::UInt64(Some(42)));
+        test_bound!(Groups, number.clone(), ScalarValue::UInt64(Some(42)));
+        test_bound!(
+            Range,
+            number.clone(),
+            ScalarValue::Utf8(Some("42".to_string()))
+        );
+
+        // Numeric (SingleQuotedString)
+        let number = Some(Box::new(ast::Expr::Value(ast::Value::SingleQuotedString(
+            "42".to_string(),
+        ))));
+        test_bound!(Rows, number.clone(), ScalarValue::UInt64(Some(42)));
+        test_bound!(Groups, number.clone(), ScalarValue::UInt64(Some(42)));
+        test_bound!(
+            Range,
+            number.clone(),
+            ScalarValue::Utf8(Some("42".to_string()))
+        );
+
+        // Non-numeric (SingleQuotedString)
+        let non_numeric = Some(Box::new(ast::Expr::Value(
+            ast::Value::SingleQuotedString("false".to_string()),
+        )));
+        test_bound_err!(
+            Rows,
+            non_numeric.clone(),
+            "Arrow error: Cast error: Cannot cast string 'false' to value of UInt64 type"
+        );
+        test_bound_err!(
+            Groups,
+            non_numeric.clone(),
+            "Arrow error: Cast error: Cannot cast string 'false' to value of UInt64 type"
+        );
+        test_bound!(
+            Range,
+            non_numeric.clone(),
+            ScalarValue::Utf8(Some("false".to_string()))
+        );
```

Comment:
Not really sure if this is expected but essentially I've tried to keep Range with the original behaviour, only changing Rows and Groups since we know they are certainly UInt64

Reply:
I guess this is alright as it is existing behaviour. 

When I tried a query with `RANGE INTERVAL 'false' PRECEDING` in `datafusion-cli` it failed like this:
```sh
type_coercion
caused by
Arrow error: Parser error: Invalid input syntax for type interval: "false"
```